geopandas/tools/tests/test_sjoin.py

```diff
@@ -926,3 +926,44 @@ def test_sjoin_nearest_inner(self):
         result5 = result5.dropna()
         result5["index_right"] = result5["index_right"].astype("int64")
         assert_geodataframe_equal(result5, result4, check_like=True)
+
+    @pytest.mark.filterwarnings("ignore:Geometry is in a geographic CRS")
+    @pytest.mark.skipif(
+        not (compat.USE_SHAPELY_20),
+        reason=(
+            "shapely >= 2.0 is required to run sjoin_nearest"
+            "with parameter `exclusive` set"
+        ),
+    )
+    def test_sjoin_nearest_exclusive(self):
+        # check equivalency of left and inner join
+
+        cities = read_file(geopandas.datasets.get_path("naturalearth_cities"))
```

Comment:
@martinfleis will likely comment on this, but my sense is that we should be avoiding new tests that rely on `geopandas.datasets`.

Since this is mostly just testing that the routing through to the correct `sindex` is working properly, you could use a very simple data frame with 2 points and sjoin to itself, just like you do with cities here.

Reply:
Thank you for the comment. I think I got carried away by the rest of the tests. I have simplified the tests now